### PR TITLE
[NETBEANS-2621] cleanup generic class iterator

### PR DESCRIPTION
**Pull request by BradWalker**
reduce the # of instances of the following error..

   [repeat] /home/bwalker/netbeans-11.0/platform/core.startup/src/org/
     netbeans/core/startup/AutomaticDependencies.java:372: warning: [rawtypes] found raw type: Iterator
   [repeat]             Iterator it = exclusions.iterator();
   [repeat]             ^
   [repeat]   missing type arguments for generic class Iterator<E>
   [repeat]   where E is a type-variable:
   [repeat]     E extends Object declared in interface Iterator

### CODE REVIEW

File: platform/openide.filesystems/src/org/openide/filesystems/MultiFileObject.java [hunk 1]
```diff
@@ -205,7 +205,6 @@ private void update() {
         }
 
         Iterator it = now.iterator();
-
         while (it.hasNext()) {
             FileObject fo = (FileObject) it.next();
```
Comment (lkishalmi) [inline platform/openide.filesystems/src/org/openide/filesystems/MultiFileObject.java]: What happened with this? Can't this converted into for as well?

Comment (BradWalker) [reply platform/openide.filesystems/src/org/openide/filesystems/MultiFileObject.java]: Unfortunately it not "low hanging fruit".. When I change this to the foreach pattern, I get the following error message..

```
   [repeat] /home/bwalker/netbeans-11.0/platform/openide.filesystems/src/org/openide/filesystems/MultiFileObject.java:207: error: incompatible types: Object cannot be converted to FileObject
   [repeat]         for (FileObject fo : now) {
   [repeat] 
```                             ^
So I left it out..

My goal is simply to get the build warnings down..

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/parser/GroovyVirtualSourceProvider.java [hunk 2]
```diff
@@ -350,7 +350,8 @@ private void genProp(PropertyNode propNode, PrintWriter out) {
             boolean skipGetter = false;
             List getterCandidates = propNode.getField().getOwner().getMethods(getterName);
             if (getterCandidates != null) {
-                for (MethodNode method : getterCandidates) {
```
Comment (matthiasblaesing) [inline groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/parser/GroovyVirtualSourceProvider.java]: Are you looking for this?

```diff
--- a/groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/parser/GroovyVirtualSourceProvider.java
+++ b/groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/parser/GroovyVirtualSourceProvider.java
@@ -348,10 +348,9 @@
             String getterName = "get" + name;
 
             boolean skipGetter = false;
-            List getterCandidates = propNode.getField().getOwner().getMethods(getterName);
+            List<MethodNode> getterCandidates = propNode.getField().getOwner().getMethods(getterName);
             if (getterCandidates != null) {
-                for (Iterator it = getterCandidates.iterator(); it.hasNext();) {
-                    MethodNode method = (MethodNode) it.next();
+                for (MethodNode method: getterCandidates) {
                     if (method.getParameters().length == 0) {
                         skipGetter = true;
                     }
```

Comment (BradWalker) [reply groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/parser/GroovyVirtualSourceProvider.java]: Yes.. It was an escape for something I was testing..

Should be reverted..

Good eyes, BTW..

**Comment by matthiasblaesing** [comment]
Merged via https://github.com/apache/netbeans/commit/fb016919734b98643341c01c5f720d205e48ecff - thank you.

**Pull request closed without merging**